fe/fe-core/src/main/java/com/starrocks/common/util/MachineInfo.java

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.nio.file.FileStore;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+
+public class MachineInfo {
+    private static final Logger LOG = LogManager.getLogger(MachineInfo.class);
+
+    /**
+     * Get the number of CPU cores.
+     *
+     * @return the number of CPU cores.
+     */
+    public static int getCpuCores() {
```

Comment:
may need to implement in a single instance mode, so the cpucore and mem info and the mac address will be only intialized once, repeated invoke of the getXXX() won't cause the FE process to parse procfs again and again giving the info are pretty immutable after process started.

Reply:
done